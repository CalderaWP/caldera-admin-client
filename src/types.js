import formType from './components/EntryViewer/formType';
import entryType from './components/EntryViewer/entryType';
import proSettingsType, {PRO_SETTINGS} from './components/Settings/ProSettings/proSettingsType';
import statusType from './components/Layout/statusType';
import privacySettingsType from './components/Settings/PrivacySettings/privacySettingsType';
import generalSettingsType from './components/Settings/GeneralSettings/generalSettingsType';

import PropTypes from 'prop-types';

/**
 * Prop types for collections of data types
 *
 * @type {{formsType: shim, entriesType: shim, proSettingsType: shim}}
 */
export const collectionTypes = {
	formsType: PropTypes.object,
	entriesType: PropTypes.object,
	settingsType: PropTypes.shape({
		[PRO_SETTINGS]: PropTypes.object,
		generalSettings: PropTypes.object,
		privacySettings: PropTypes.object
	})

};

/**
 * Prop types for singular data types.
 */
export default {
	formType,
	entryType,
	proSettingsType,
	generalSettingsType,
	privacySettingsType,
	statusType
}
