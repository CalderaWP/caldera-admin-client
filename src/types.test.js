import types,{collectionTypes} from "./types";


describe('types',() => {
	it( 'has the types', () => {
		expect(JSON.stringify(types)).toMatchSnapshot();

	});

	it( 'has the collectionTypes', () => {
		expect(JSON.stringify(collectionTypes)).toMatchSnapshot();

	});

});