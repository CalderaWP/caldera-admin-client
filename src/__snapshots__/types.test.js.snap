// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`types has the collectionTypes 1`] = `"{}"`;

exports[`types has the types 1`] = `"{\\"formType\\":{},\\"entryType\\":{},\\"proSettingsType\\":{},\\"privacySettingsType\\":{}}"`;
